src/server/debug/server/server.go

```diff
@@ -927,3 +946,104 @@ func redirectDump(ctx context.Context, c debug.DebugClient, filter *debug.Filter
 	}
 	return grpcutil.NewStreamingBytesReader(dumpC, nil), nil
 }
+
+func (s *debugServer) collectDatabaseStats(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	if err := s.collectDatabaseActivities(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseRowCount(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseTableSizes(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseActivities(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "activities", s.queryDatabaseActivities, prefix...)
+}
+
+func (s *debugServer) collectDatabaseRowCount(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "row-counts", s.queryDatabaseRowCounts, prefix...)
+}
+
+func (s *debugServer) collectDatabaseTableSizes(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "table-sizes", s.queryDatabaseTableSizes, prefix...)
+}
+
+func (s *debugServer) collectDatabaseQuery(ctx context.Context, tw *tar.Writer, fileName string,
+	queryFunc func(ctx context.Context, db *pachsql.DB, w io.Writer) error, prefix ...string) error {
+	return collectDebugFile(tw, fileName, "json", func(w io.Writer) error {
+		if err := queryFunc(ctx, s.database, w); err != nil {
+			return errors.EnsureStack(err)
+		}
+		return nil
+	}, prefix...)
+}
+
+func (s *debugServer) queryDatabaseRowCounts(ctx context.Context, db *pachsql.DB, w io.Writer) error {
+	query := `
+		SELECT schemaname, relname, n_live_tup, seq_scan, idx_scan
+		FROM pg_stat_user_tables
+		ORDER BY schemaname, relname;
+	`
+	return s.queryDatabase(ctx, query, w)
+}
+
+func (s *debugServer) queryDatabaseActivities(ctx context.Context, db *pachsql.DB, w io.Writer) error {
+	query := `
+		SELECT current_timestamp - query_start as runtime, datname, usename, client_addr, query
+		FROM pg_stat_activity
+		WHERE state != 'idle'
+		ORDER by runtime DESC;
+	`
+	return s.queryDatabase(ctx, query, w)
+}
+
+func (s *debugServer) queryDatabaseTableSizes(ctx context.Context, db *pachsql.DB, w io.Writer) error {
+	query := `
+		SELECT nspname || '.' || relname AS "relation",
+		  pg_size_pretty(pg_total_relation_size(C.oid)) AS "total_size"
```

Comment:
I'd rather we didn't make the sizes pretty here.  I think an integer number of bytes is easier to work with.

Reply:
Can do